io/plugins/eu.esdihumboldt.hale.io.gml/test/eu/esdihumboldt/hale/io/gml/reader/internal/GmlInstanceCollectionTest.java

```diff
@@ -262,6 +272,58 @@ public void testLoadShiporder() throws Exception {
 		}
 	}
 
+	/**
+	 * Test loading a simple XML file with one instance
+	 *
+	 * @throws Exception if an error occurs
+	 */
+	@Test
+	public void testLoadFilterNoDuplicates() throws Exception {
+
+		int uniqueInstanceCount = extractedUniqueInstances(
+				"/data/filter_doubles/filter_with_doubles.xml");
+		System.out.println(
+				"There are " + uniqueInstanceCount + " unique instances in the file with doubles.");
+		assertEquals(3, uniqueInstanceCount);
+
+		uniqueInstanceCount = extractedUniqueInstances(
+				"/data/filter_doubles/filter_no_doubles.xml");
+
+		System.out.println("There are " + uniqueInstanceCount
+				+ " unique instances in the file with unique instances.");
+		assertEquals(4, uniqueInstanceCount);
+	}
+
+	/**
+	 * @param pathToXML
+	 * @return how many unique instances are present
+	 * @throws IOException
+	 * @throws IOProviderConfigurationException
+	 * @throws URISyntaxException
+	 */
+	private int extractedUniqueInstances(String pathToXML)
+			throws IOException, IOProviderConfigurationException, URISyntaxException {
+		GmlInstanceCollection instances = loadInstances(
+				getClass().getResource("/data/filter_doubles/filter.xsd").toURI(),
+				getClass().getResource(pathToXML).toURI(), false);
+
+		FullInstanceIteratorSupport fullInstance = new FullInstanceIteratorSupport(
+				instances.iterator());
```

Comment:
see [my comment here](https://github.com/halestudio/hale/pull/1180/files#r1725319325)

Reply:
Removed the use of FullInstanceIteratorSupport here.